src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -42,32 +52,104 @@ protected override async Task AddCompletionItemsAsync(CompletionContext completi
+        {
+            // In case the caret is at the beginning of the file, take the root node.
+            var aliasDeclarations = GetAliasDeclarationNodes(
+                syntaxContext.LeftToken.Parent ?? syntaxContext.SyntaxTree.GetRoot(cancellationToken));
+            foreach (var aliasNode in aliasDeclarations)
+            {
+                var symbol = syntaxContext.SemanticModel.GetDeclaredSymbol(aliasNode, cancellationToken);
+                if (symbol is IAliasSymbol {Target: ITypeSymbol target})
+                {
+                    yield return target;
+                }
+            }
+        }
+
+        private static void AddItems(
+            ImmutableArray<CompletionItem> items,
+            CompletionContext completionContext,
+            HashSet<string> namespacesInScope,
+            MultiDictionary<string, string> aliasTargetNamespaceToTypeNameMap,
+            TelemetryCounter counter)
         {
             counter.ReferenceCount++;
             foreach (var item in items)
             {
-                var containingNamespace = ImportCompletionItem.GetContainingNamespace(item);
-                if (!namespacesInScope.Contains(containingNamespace))
+                if (ShouldAddItem(item, namespacesInScope, aliasTargetNamespaceToTypeNameMap))
                 {
                     // We can return cached item directly, item's span will be fixed by completion service.
                     // On the other hand, because of this (i.e. mutating the  span of cached item for each run),
-                    // the provider can not be used as a service by components that might be run in parallel 
+                    // the provider can not be used as a service by components that might be run in parallel
                     // with completion, which would be a race.
                     completionContext.AddItem(item);
                     counter.ItemsCount++;
                 }
             }
+
+            static bool ShouldAddItem(
+                CompletionItem item,
+                HashSet<string> namespacesInScope,
+                MultiDictionary<string, string> aliasTargetNamespaceToTypeNameMap)
+            {
+                var containingNamespace = ImportCompletionItem.GetContainingNamespace(item);
+                // 1. if the namespace of the item is in scoop. Don't add the item
+                if (namespacesInScope.Contains(containingNamespace))
+                {
+                    return false;
+                }
+
+                // 2. If the item might be an alias target. First check if its namespace is in the map.
+                if (aliasTargetNamespaceToTypeNameMap.ContainsKey(containingNamespace))
+                {
+                    // Then check its type name.
+                    // It is done in this way because we don't want to get type name for all the items
+                    var typeName = ImportCompletionItem.GetTypeName(item);
```

Comment:
This gives you the metadata name like "System.Collections.Generic.Dictionary`2", not sure how it would work if you try to match the name from a instantiated generic like `System.Collections.Generic.Dictionary<int, int>`.